### PR TITLE
first create pass refactoring

### PR DESCRIPTION
**Pull request by pkozlowski-opensource**

### CODE REVIEW

**Review by mhevery — APPROVED** [review]

**Comment by mhevery** [comment]
[presubmit](http://test/OCL:281114489:BASE:281114531:1574105359842:927ee4d)
[ivy presubmit](http://test/OCL:281114489:BASE:281114709:1574105407427:fc950148)

File: packages/core/src/render3/instructions/text.ts [hunk 1]
```diff
@@ -25,14 +25,21 @@ import {getOrCreateTNode} from './shared';
...
                    'text nodes should be created before any bindings');
-  ngDevMode && assertDataInRange(lView, index + HEADER_OFFSET);
-  const textNative = lView[index + HEADER_OFFSET] = createTextNode(value, lView[RENDERER]);
-  const tNode = getOrCreateTNode(lView[TVIEW], lView[T_HOST], index, TNodeType.Element, null, null);
+  ngDevMode && assertDataInRange(lView, adjustedIndex);
 
-  // Text nodes are self closing.
-  setIsNotParent();
+  const tNode = tView.firstCreatePass ?
+      getOrCreateTNode(tView, lView[T_HOST], index, TNodeType.Element, null, null) :
```
Comment (kara) [inline packages/core/src/render3/instructions/text.ts]: Is the longer term plan to remove the conditional from `getOrCreateTNode` so we aren't reading the `tNode` from the `TView` when we know it hasn't been created? i.e. this conditional: 
```ts
tView.data[adjustedIndex] as TNode ||
      createTNodeAtIndex(tView, tHostNode, adjustedIndex, type, name, attrs);
```
From this PR, it looks like we're slowly making it into `createTNode` without the `get` part

**Review by kara — APPROVED** [review]
LGTM

File: packages/core/src/render3/instructions/text.ts (hunk 1 above)
Comment (kara) [reply packages/core/src/render3/instructions/text.ts]: Nevermind, just saw https://github.com/angular/angular/pull/33856/files#diff-b308ea52d8ee98b711ae369515139f08R54 :)   We are on the same page!

Comment (pkozlowski-opensource) [reply packages/core/src/render3/instructions/text.ts]: Yes and yes @kara :-) Inching towards the `createTnode` (which will allow us to to cleanups in other places, too!)

**Pull request closed without merging**